orchagent/vnetorch.cpp

```diff
@@ -249,6 +259,550 @@ VNetVrfObject::~VNetVrfObject()
+    SWSS_LOG_ENTER();
+
+    return false;
+}
+
+uint32_t VNetBitmapObject::getFreeBitmapId(const string& vnet)
+{
+    SWSS_LOG_ENTER();
+
+    for (uint32_t i = 0; i < VNET_BITMAP_SIZE; i++)
+    {
+        uint32_t id = 1 << i;
+        if ((id & vnetBitmap_) == 0)
+        {
+            vnetBitmap_ |= id;
+            vnetIds_.emplace(vnet, id);
+            return id;
+        }
+    }
+
+    return 0;
+}
+
+uint32_t VNetBitmapObject::getBitmapId(const string& vnet)
+{
+    SWSS_LOG_ENTER();
+
+    if (vnetIds_.find(vnet) == vnetIds_.end())
+    {
+        return 0;
+    }
+
+    return vnetIds_[vnet];
+}
+
+void VNetBitmapObject::recycleBitmapId(uint32_t id)
+{
+    SWSS_LOG_ENTER();
+
+    vnetBitmap_ &= ~id;
+}
+
+uint32_t VNetBitmapObject::getFreeTunnelRouteTableOffset()
+{
+    SWSS_LOG_ENTER();
+
+    for (uint32_t i = 0; i < 256; i++)
+    {
+        if (tunnelOffsets_.count(i) == 0)
+        {
+            tunnelOffsets_.insert(i);
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+void VNetBitmapObject::recycleTunnelRouteTableOffset(uint32_t offset)
+{
+    SWSS_LOG_ENTER();
+
+    tunnelOffsets_.erase(offset);
+}
+
+VnetBridgeInfo VNetBitmapObject::getBridgeInfoByVni(uint32_t vni, string tunnelName)
+{
+    SWSS_LOG_ENTER();
+
+    if (bridgeInfoMap_.find(vni) != bridgeInfoMap_.end())
+    {
+        return std::move(bridgeInfoMap_.at(vni));
+    }
+
+    sai_status_t status;
+    VnetBridgeInfo info;
+    sai_attribute_t attr;
+    vector<sai_attribute_t> bridge_attrs;
+    attr.id = SAI_BRIDGE_ATTR_TYPE;
+    attr.value.s32 = SAI_BRIDGE_TYPE_1D;
+    bridge_attrs.push_back(attr);
+
+    status = sai_bridge_api->create_bridge(
+            &info.bridge_id,
+            gSwitchId,
+            (uint32_t)bridge_attrs.size(),
+            bridge_attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create bridge for vni %u", vni);
+        throw std::runtime_error("vni creation failed");
+    }
+
+    vector<sai_attribute_t> rif_attrs;
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_VIRTUAL_ROUTER_ID;
+    attr.value.oid = gVirtualRouterId;
+    rif_attrs.push_back(attr);
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_SRC_MAC_ADDRESS;
+    memcpy(attr.value.mac, gMacAddress.getMac(), sizeof(sai_mac_t));
+    rif_attrs.push_back(attr);
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_TYPE;
+    attr.value.s32 = SAI_ROUTER_INTERFACE_TYPE_BRIDGE;
+    rif_attrs.push_back(attr);
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_BRIDGE_ID;
+    attr.value.oid = info.bridge_id;
+    rif_attrs.push_back(attr);
+
+    status = sai_router_intfs_api->create_router_interface(
+            &info.rif_id,
+            gSwitchId,
+            (uint32_t)rif_attrs.size(),
+            rif_attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create rif for vni %u", vni);
+        throw std::runtime_error("vni creation failed");
+    }
+
+    SWSS_LOG_NOTICE("Created RIF");
```

Comment:
Since it is notice, please add more info like vni, tunnel name. If not relevant, suggest to move to INFO.

Reply:
Just a debug comment. Removed